Preparing for resident

aborsu opened the pull request.

Commit:
fix(spelling): Consistent case folding for KGB

Commit:
refactor(kgitb): Shorten folder name

Commit:
refactor(deps): Move flask out of init

Commit:
refactor(py2): Render compatible

Commit:
test(switch): Add 2&3 cross test

SagaStalin commented on kgitb/config.py:
```diff
@@ -1,8 +1,9 @@
+# -*- coding: utf-8 -*-
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Missing module docstring [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0111)

SagaStalin commented on kgitb/config.py:
```diff
@@ -1,8 +1,9 @@
+# -*- coding: utf-8 -*-
 from os import environ as env
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a docstring to this module. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)

SagaStalin commented on kgitb/__main__.py:
```diff
@@ -1,12 +1,17 @@
 import logging
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a docstring to this module. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Missing module docstring [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0111)

SagaStalin commented on kgitb/__main__.py:
```diff
@@ -1,12 +1,17 @@
+from .config import GH_TOKEN, HOST, PORT, HIDDEN
+from . import routes
 
 log = logging.getLogger(__name__)
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Invalid constant name "log" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

SagaStalin commented on kgitb/__main__.py:
```diff
@@ -1,12 +1,17 @@
 log = logging.getLogger(__name__)
 
 
+app = routes.setup(Flask(HIDDEN[1:-1]))
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Invalid constant name "app" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

SagaStalin commented on kgitb/__main__.py:
```diff
@@ -1,12 +1,17 @@
+app = routes.setup(Flask(HIDDEN[1:-1]))
+
+
 def main():
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a docstring to this function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Missing function docstring [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0111)

SagaStalin commented:
SonarQube analysis reported 102 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 50 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 52 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [__init__.py#L1](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/__init__.py#L1): Add a docstring to this module. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [github.py#L14](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L14): Add a docstring to this function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [github.py#L15](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L15): Remove this "\", add another "\" to escape it, or make this a raw string. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1717)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [github.py#L18](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L18): Add a docstring to this function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [github.py#L22](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L22): Add a docstring to this function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [github.py#L27](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L27): Add a docstring to this function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [github.py#L36](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L36): Add a docstring to this function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [github.py#L40](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L40): Add a docstring to this function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [github.py#L59](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/f8f655718998f74c4849c9ba0c91fd63ccc31399/kgitb/github.py#L59): Add a docstring to this function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1720)

Commit:
style(sagastalin): Comply

SagaStalin commented on kgitb/rules/status_rules.py:
```diff
@@ -4,9 +4,9 @@
 from ..config import TYPES
 
 status_pattern = re.compile(
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Invalid constant name "status_pattern" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

SagaStalin commented on kgitb/rules/status_rules.py:
```diff
@@ -4,9 +4,9 @@
+    r"^([^()\s]*)(\s*)\(([^()\s]*)\)(\s*):(\s*)(.*\w)(\s*)$")
 
-spaces = re.compile("\w*\s+\w*\(|\w+\(.*[^\w].*\):")
+spaces = re.compile(r"\w*\s+\w*\(|\w+\(.*[^\w].*\):")
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Invalid constant name "spaces" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

SagaStalin commented on kgitb/rules/status_rules.py:
```diff
@@ -4,9 +4,9 @@
-spaces = re.compile("\w*\s+\w*\(|\w+\(.*[^\w].*\):")
+spaces = re.compile(r"\w*\s+\w*\(|\w+\(.*[^\w].*\):")
 
 _type_string = ", ".join(sorted(TYPES))
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Invalid constant name "_type_string" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

SagaStalin commented:
SonarQube analysis reported 24 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 10 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 14 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [__init__.py#L61](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/__init__.py#L61): Remove the parentheses after this "return" keyword [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1721)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [__init__.py#L61](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/__init__.py#L61): Remove those useless parentheses [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1110)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [status_rules.py#L27](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L27): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1066)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [status_rules.py#L31](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L31): This function has 4 returns or yields, which is more than the 3 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [status_rules.py#L31](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L31): Function has a complexity of 15 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [status_rules.py#L65](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L65): Remove those useless parentheses [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1110)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [status_rules.py#L65](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b136e4825f3d0305fdfd00c75368b6eac3b7d11d/kgitb/rules/status_rules.py#L65): Remove the parentheses after this "if" keyword [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1721)

SagaStalin commented on kgitb/__main__.py:
```diff
@@ -0,0 +1,21 @@
+if __name__ == '__main__':
+    if GH_TOKEN is None:
+        sys.exit("GH_TOKEN must be set.")
+    LOG.basicConfig(level=logging.DEBUG)
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Instance of 'RootLogger' has no 'basicConfig' member [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AE1101)

SagaStalin commented on kgitb/rules/__init__.py:
```diff
@@ -38,13 +38,19 @@ def _get_rules(module):
     return re.split('\r?\n', commit_message)
 
 
 def apply(commit_message):
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Redefining built-in 'apply' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AW0622)

SagaStalin commented on kgitb/rules/__init__.py:
```diff
@@ -38,13 +38,19 @@ def _get_rules(module):
 _status_rules = _get_rules(status_rules)
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Invalid constant name "_status_rules" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

SagaStalin commented:
SonarQube analysis reported 12 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 5 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 7 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [__init__.py](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/__init__.py): Add a new line at the end of this file "__init__.py". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS113)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [github.py#L11](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/github.py#L11): Invalid constant name "log" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [routes.py#L6](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/routes.py#L6): Invalid constant name "log" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [__init__.py#L36](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/rules/__init__.py#L36): Invalid constant name "_line_rules" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [__init__.py#L37](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/b0d82a4088ed5a2c058eb7472e92cc3bec636a8f/kgitb/rules/__init__.py#L37): Invalid constant name "_raw_rules" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

SagaStalin commented on kgitb/rules/status_rules.py:
```diff
@@ -23,9 +17,8 @@ def check_status_length(status_line):
+        return "Limit status to {0} characters".format(MAX_STATUS_LENGTH)
 
 
 def check_status_formatting(status_line):
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has a complexity of 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)

Commit:
feat(resident): Install

Commit:
feat(resident): Add message linter

Commit:
feat(resident): Add correct message

Commit:
fix(packaging): Add License and rules

Commit:
fix(trim_status): Incompatible with hooks

Git automatically trims new lines when commiting the message
but also includes them in the commit file.
The rule is therefore unneeded and counter-productive.

Commit:
fix(errors): Errors are empty not None

Commit:
fix(hook): Make executable

Commit:
test(sonar): Check scripts

Commit:
refactor(packaging): Add classifiers and keywords

Commit:
fix(hook): Add absolute path

Sourcetree could not find the resident,
so resident adds a full path to itself.

Commit:
fix(resident): Don't overwrite hook

Resident kept overriding the hook if it didn't find itself in it.
Also fixed the way it tries to find itself.

Commit:
fix(arguments): Remove debug message

Commit:
doc(readme): Add resident and lubyanka info

Bump version to rerelease on pypitest

Commit:
fix(packaging): Prettier description

kxepal commented on setup.py:
```diff
@@ -1,31 +1,25 @@
+    version='0.0.6',
     description='A commit message linter',
-    long_description=read_md('README.md'),
+    long_description=io.open(readme, encoding="utf-8").read(),
```
This will cause resource warnings for Python 3 since you didn't close the file object.

Commit:
test(fix): Rename old rule

kxepal commented on bin/resident:
```diff
@@ -0,0 +1,202 @@
+    else:
+        result = True
+        for file_path, file_errors in zip(file_paths, errors):
+            if len(file_errors) != 0:
```
`if file_errors` is more Pythonic. Same for above `len(errors[0]) != 0`.

kxepal commented on bin/resident:
```diff
@@ -0,0 +1,202 @@
+        for file_path, file_errors in zip(file_paths, errors):
+            if len(file_errors) != 0:
+                format_errrors(file_errors, file_path)
+                result = False
```
If result set to False, may be break out the loop then?

aborsu replied:
I thought i would check all the files and comment on all of them.
Avoids having to fix one file and then having the test fail on the following one?

kxepal replied:
Hm...no, that's a good case.

UPD. I mean, it's good to check all the files and report all the stuff in single shot.

kxepal commented on setup.py:
```diff
@@ -0,0 +1,33 @@
+    scripts=['bin/resident'],
+    version='0.0.6',
+    description='A commit message linter',
+    long_description=io.open(readme, encoding="utf-8").read(),
```
This will cause resource warnings for Python 3 since you didn't close the file object.

aborsu commented on bin/resident:
```diff
@@ -0,0 +1,202 @@
+        for file_path, file_errors in zip(file_paths, errors):
+            if len(file_errors) != 0:
+                format_errrors(file_errors, file_path)
+                result = False
```
No, the loop copies the old file to a temp file that overwrites the file.
(I couldn't find any better way to modify a line in a file in place).
So if we stop the loop, the new file will be truncated.

This is to play nice with other hooks that might exist.

Commit:
fix(sonar): Switch -> java8

Commit:
fix(packaging): Py3 unclosed fh warning

SagaStalin commented:
SonarQube analysis reported 11 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 5 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 6 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [__init__.py](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/__init__.py): Add a new line at the end of this file "__init__.py". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS113)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [github.py#L11](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/github.py#L11): Invalid constant name "log" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [routes.py#L6](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/routes.py#L6): Invalid constant name "log" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [__init__.py#L36](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/rules/__init__.py#L36): Invalid constant name "_line_rules" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [__init__.py#L37](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/3f9c173402da6d1967fe0b17a9fca9669ba934f0/kgitb/rules/__init__.py#L37): Invalid constant name "_raw_rules" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

Commit:
fix(setup_hook): Use given path

SagaStalin commented:
SonarQube analysis reported 11 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 5 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 6 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [__init__.py](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/__init__.py): Add a new line at the end of this file "__init__.py". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS113)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [github.py#L11](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/github.py#L11): Invalid constant name "log" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [routes.py#L6](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/routes.py#L6): Invalid constant name "log" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [__init__.py#L36](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/rules/__init__.py#L36): Invalid constant name "_line_rules" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [__init__.py#L37](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/7c8affc91142ff23218fdef97af6ab602b8e834f/kgitb/rules/__init__.py#L37): Invalid constant name "_raw_rules" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

Commit:
feat(resident): Uninstall method

Commit:
refactor(soft_tabs): As line rule

Commit:
feat(comments): Ignore git comments

Commit:
version(0.0.8): Pypitest

ghost commented:
SonarQube analysis reported 11 issues
- ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 5 major
- ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 6 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [config.py#L25](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/config.py#L25): Make this IP "0.0.0.0" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1313)
2. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/routes.py#L34): Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AFunctionComplexity)
3. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [routes.py#L34](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/routes.py#L34): This function has 6 returns or yields, which is more than the 3 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS1142)
4. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [**init**.py](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/__init__.py): Add a new line at the end of this file "**init**.py". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS113)
5. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [github.py#L11](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/github.py#L11): Invalid constant name "log" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)
6. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [routes.py#L6](https://github.com/Sagacify/komitet-gita-bezopasnosti/blob/cb255eccb53a72a0c391201ffbe9651bdcf4dcd4/kgitb/routes.py#L6): Invalid constant name "log" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

SagaStalin commented on kgitb/rules/__init__.py:
```diff
@@ -34,22 +34,25 @@ def _get_rules(module):
             if tup[0][0] != "_"]
 
 _line_rules = _get_rules(line_rules)
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Invalid constant name "_line_rules" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AC0103)

SagaStalin commented on kgitb/rules/__init__.py:
```diff
@@ -34,22 +34,25 @@ def _get_rules(module):
+    if _is_merge(commit_message):
+        return errors
     commit_lines = split_lines(commit_message)
+    status = commit_lines[0]
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Unused variable 'status' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=Pylint%3AW0612)
